### PR TITLE
slight modification of step-30

### PR DESCRIPTION
**Pull request by gioacchinowang**
The active cell selection rule doesn't work in 1D problems,
but works by a slight modification.
The point is, in a 1D problem, there are no actual sub-surfaces, a surface can not see
any refined neighbor surfaces.
The old selection rule would easily choose a non-active neighbor, but the new rule can avoid it.

### CODE REVIEW

**Comment by drwells** [comment]
I don't think these are equivalent in 2D: I added some code to step-30 to print cell ids in case c and I got different results with this patch and on master.

Nevertheless, this looks reasonable (and the graphical output looks the same): perhaps the difference is not important. I'll try to find some time to look over this today or tomorrow to see if I can convince myself that this is the right change.

File: examples/step-30/step-30.cc [hunk 1]
```diff
@@ -568,13 +568,13 @@ namespace Step30
                     // situation of our cell and the neighbor we want to treat
                     // only one half, so that each face is considered only
                     // once. Thus we have the additional condition, that the
-                    // cell with the lower index does the work. In the rare
-                    // case that both cells have the same index, the cell with
-                    // lower level is selected.
+                    // cell with the higher level does the work. In the rare
+                    // case that both cells have the same level, the cell with
```
Comment (bangerth) [inline examples/step-30/step-30.cc]: Just drop "In the rare case" and say "If both cells have the same level"

File: examples/step-30/step-30.cc [hunk 2]
```diff
@@ -568,13 +568,13 @@ namespace Step30
...
+                    // lower index is selected.
                     if (!cell->neighbor_is_coarser(face_no) &&
-                        (neighbor->index() > cell->index() ||
-                         (neighbor->level() < cell->level() &&
-                          neighbor->index() == cell->index())))
+                        (neighbor->level() < cell->level() ||
```
Comment (bangerth) [inline examples/step-30/step-30.cc]: This condition can never be correct. You are saying
   if (neighbor is not coarser
       &&
       (neighbor is coarser   || other conditions...)
I think you want a different condition here :-)

**Comment by drwells** [comment]
@gioacchinowang Let us know if you need help with the indentation :)

**Comment by gioacchinowang** [comment]
Thanks for reminding David,
I've followed the instructions :-)

Jiaxin Wang
--------------
AstroParticlePhys,
SISSA, Italy

**Comment by drwells** [comment]
I think you need to move `continue` to its own line.

File: examples/step-30/step-30.cc [hunk 3]
```diff
@@ -576,42 +576,42 @@ namespace Step30
                               (neighbor->index() > cell->index() &&
```
Comment (drwells) [inline examples/step-30/step-30.cc]: I believe the strange indentation is a result of a missing parenthesis on line 574.

**Comment by gioacchinowang** [comment]
Yes I got it, thanks!

**Comment by drwells** [comment]
@bangerth I would like to merge this (well, we should rebase first) and I am pretty sure the changes satisfy the comments left in your review. Are you happy with this?

**Comment by bangerth** [comment]
Let's do this -- my apologies for sitting on this for so long. Thanks, @gioacchinowang, for writing the patch!

I've been entirely confused by this code trying to read over it a number of times now. It is in desperate need of more explanations. Follow-up patch coming in soon.

**Pull request merged**